public/dorequest.php

```diff
@@ -30,12 +34,14 @@
     $validLogin = authenticateFromAppToken($user, $token, $permissions);
 }
 
-function DoRequestError(string $error): JsonResponse
-{
-    return response()->json([
-        'Success' => false,
-        'Error' => $error,
-    ]);
+if (!function_exists('DoRequestError')) {
```

Comment:
Out of curiosity - did we ever run into this function being defined more than once?

Reply:
I added this logic as part of this PR. The unit tests fail without it:
```
PHP Fatal error:  Cannot redeclare DoRequestError() (previously declared in
/home/brian/source/RAWeb/public/dorequest.php:38) in /home/brian/source/RAWeb/public/dorequest.php on line 38
```

I couldn't figure out where the extra reference is coming from, so I just wrapped it in this logic.

Reply:
Might have to do how the tests are executed in combination with the include logic for script files in `HandlesPublicFileRequests`.
We can ignore that for now as those includes shall be rewritten to controller actions at some point.